api/candlepin-api-spec.yaml

```diff
@@ -284,6 +284,86 @@ components:
         href:
           type: string
 
+    ActivationKeyDTO:
+      description: A token that allows registration without username/password and sets subscription and content related properties at registration.
+      allOf:
+        - $ref: '#/components/schemas/TimestampedEntity'
+        - type: object
+          properties:
+            id:
+              type: string
+            name:
+              type: string
```

Comment:
Just an idea. We could add pattern: "^[a-zA-Z0-9_-]+$" so that we don't have to validate the name manually.

Reply:
Does the pattern method basically add a regex validator on this field? Does it also handle nulls?

Reply:
I just realized that the resource is also doing this:
```
        if (StringUtils.isBlank(dto.getName())) {
            throw new BadRequestException(i18n.tr("Must provide a name for activation key."));
        }
```
so I can simply add `name` to the list of required properties (which will generate a `@NotNull` annotation) and will validate automatically

Reply:
Exactly. It just adds @Pattern annotation. @Pattern is not validated if the value is null.

Reply:
I decided not to do this because it will change the current error message we send back to [this one](https://github.com/candlepin/candlepin/blob/32258d638e79e90eba5cf4cb5b3efc12354ca24b/common/src/main/java/org/candlepin/common/validation/CandlepinMessageInterpolator.java#L69), which is less user friendly.